scripts/locale-checks.ts

```diff
@@ -0,0 +1,318 @@
+  ...WEEKDAY_CYCLES.map((c) => `cycles:weekday_${c.toLowerCase()}`),
+  ...PSALTER_WEEKS.map((c) => `cycles:psalter_${c.toLowerCase()}`),
+  ...RANKS.map((r) => `ranks:${r.toLowerCase()}`),
+  ...MONTHS.map((_m, i) => `months:${i}`),
+  ...WEEKDAYS.map((_w, i) => `weekdays:${i}`),
+];
+
+/**
+ * Compute data to check
+ */
+for (let i = 0; i < allCalendars.length; i++) {
+  const calendar = allCalendars[i];
+
+  // Init config
+  const isGRC = calendar.name === GeneralRoman.name;
+  const builder = isGRC ? new RomcalBuilder(devLocale) : new RomcalBuilder(devLocale, calendar);
+
+  const inputs = builder.getAllInputs();
+  const definitions = builder.getAllDefinitions();
+
+  // Retrieve martyrology computed keys from definitions
+  const martyrologyComputedKeysFromDefinitions = Object.entries(definitions).reduce<StringSetRecord>(
+    (acc, [key, value]) => {
+      const mKeys = value.reduce<string[]>((vAcc, definition) => {
+        return [
+          ...vAcc,
+          ...(definition.martyrology?.reduce<string[]>((mAcc, m) => {
+            if (typeof m === 'string') return [...mAcc, m];
+            return [...mAcc, m.id];
+          }, []) ?? []),
+        ];
+      }, []);
+      if (mKeys.length) {
+        acc[key] = new Set<string>();
+        mKeys.forEach((mKey) => acc[key].add(mKey));
+      }
+      return acc;
+    },
+    {},
+  );
+
+  // Now, retrieve martyrology keys evaluated from inputs, to combine them with the ones from definitions.
+  // This is necessary because the inputs may have dropped some martyrology keys (when a martyrology key is not found).
+  const martyrologyComputedKeys = Object.values(inputs).reduce<Set<string>>((acc, value) => {
+    const mkeys = martyrologyComputedKeysFromDefinitions[value.id]
+      ? martyrologyComputedKeysFromDefinitions[value.id]
+      : value.martyrology?.map((m) => (typeof m === 'string' ? m : m.id));
+    mkeys.forEach((mkey) => acc.add(mkey));
+
+    return acc;
+  }, new Set<string>());
+
+  // Retrieve locale computed keys from inputs
+  const localeComputedKeys = Object.values(inputs).reduce<string[]>(
+    (acc, value) => [...acc, value.i18nDef[0]],
+    metaI18nKeys,
+  );
+
+  allCalendarData[calendar.name] = {
+    localeComputedKeys,
+    martyrologyComputedKeys: Array.from(martyrologyComputedKeys),
+  };
+
+  localeComputedKeys.forEach((key) => allLocalesKeys.add(key));
+  martyrologyComputedKeys.forEach((key) => allMartyrologyKeys.add(key));
+}
+
+/**
+ * [1] If there are useless martyrology items.
+ */
+const uselessMartyrologyKeys = findMissingInArray(Object.keys(Martyrology.catalog), allMartyrologyKeys);
+logIf('error', `Useless martyrology items:`, uselessMartyrologyKeys);
+
+/**
+ * [2] If there are missing martyrology items.
+ */
+const missingMartyrologyKeys = findMissingInArray(allMartyrologyKeys, Object.keys(Martyrology.catalog));
+logIf('error', `Missing martyrology items:`, missingMartyrologyKeys);
+
+/**
+ * [3] If there are missing localized 'en' items
```

Comment:
Nitpick: I would remove the trailing dots (`.`) from short code comments, however, I definitely would try to be consistent → either add a dot here or remove them from other short code comments.

Reply:
👍 I prefer to keep the trailing dot everywhere.